### DISCUSSION TITLE
How to start WiFi Manager without disturbing the mode

### DISCUSSION THREAD

**Discussion by ramanraja** [arendst/Tasmota#14141] [Q&A] [answered]
I wanted to start the Wifi Manager when I long press a button on my Tasmota device.
So I setup the following rule:
`on button1#state=3 do wificonfig 2 endon`
This starts the Wifi Manager as desired, but it also has the side effect of permanently changing the `wificonfig `mode to 2.
The disadvantage is, it keeps rebooting the device every 3 minutes if the AP is not available.
I want the wificonfig mode to remain its default 4, RETRY mode. 
Of course, I can issue the command `wificonfig 4 `manually. But this is not an option for me, as the device is to be used by non-technical clients.
Is there a way to just start the WiFi Manager only on demand, without changing the mode?

**Comment by sfromis** [comment, marked as answer]
6 short clicks on a pin configured as a Button should start wifi manager mode, unless multipress disabled.

**Comment by ramanraja** [comment]
OK, this works for me. Though I was hoping for a command like 
`start WifiManager ` :)
Thanks

**Reply by coolman111** [reply]
Hello ramanraja,
I think, your main problem is not resolved with the above answers.
When you start WiFi Manager, it does not matter by which method, on the exit of WiFi Manager, the wificonfig is remaining on 2, anyway.
The only way is to set it back to 4 by a rule, because on exiting WiFi Manager, Tasmota reboots the device.
So, if you have your rule:
on button1#state=3 do wificonfig 2 endon
you add an additional command to this rule:
on system#boot do wificonfig 4 endon
So your rule1 looks as follows:
rule1 on button1#state=3 do wificonfig 2 endon on system#boot do wificonfig 4 endon

That's all. Now wificonfig 4 will be set on every exit of WiFi Manager.

Another hint:
I noticed that since Tasmota 12.x you cannot leave WiFi Manager by a simple 1 click of the button as in former releases.
You now have to click 6 times again or wait the timeout (3 minutes).
I don't know if this is a bug or a changed feature.

**Reply by ramanraja** [reply]
This is useful hint. I have implemented this basic idea within my app:  (1) The user long-presses a button and the Wi-Fi Manager starts. (2) The user enters the Wi-Fi credentials on the mobile app and submits. (3) The mobile app sends the credentials to Tasmota, and as the last instruction, has your `wificonfig 4`. This has solved my problem. Thank you.
